iroha-cli/interactive/impl/interactive_custom_action_cli.cpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "interactive/interactive_custom_action_cli.hpp"
+#include <client.hpp>
+
+namespace iroha_cli {
+  namespace interactive {
+
+    InteractiveCustomActionCli::InteractiveCustomActionCli() {
+      createActionsMenu();
+      createResultMenu();
+    }
+
+    void InteractiveCustomActionCli::createActionsMenu() {
+      descriptionMap_ = {{GET_TX_INFO, "Get status of transaction"}};
+      const auto tx_id = "Requested tx hash";
+
+      requestParamsDescriptions_ = {{GET_TX_INFO, {tx_id}}};
+      actionHandlers_ = {
+          {GET_TX_INFO, &InteractiveCustomActionCli::parseGetHash}};
+
+      menuPoints_ = formMenu(
+          actionHandlers_, requestParamsDescriptions_, descriptionMap_);
+      addBackOption(menuPoints_);
+    }
+
+    void InteractiveCustomActionCli::createResultMenu() {
+      resultHandlers_ = {
+          {SEND_CODE, &InteractiveCustomActionCli::parseSendToIroha},
+          {SAVE_CODE, &InteractiveCustomActionCli::parseSaveFile}};
+      resultParamsDescriptions_ = getCommonParamsMap();
+
+      resultPoints_ = formMenu(resultHandlers_,
+                               resultParamsDescriptions_,
+                               getCommonDescriptionMap());
+      addBackOption(resultPoints_);
+    }
+
+    void InteractiveCustomActionCli::run() {
+      bool isParsing = true;
+      currentContext_ = MAIN;
+      printMenu("Choose action: ", menuPoints_);
+      while (isParsing) {
+        auto line = promtString("> ");
+        switch (currentContext_) {
+          case MAIN:
+            isParsing = parseAction(line);
+            break;
+          case RESULT:
+            isParsing = parseResult(line);
+            break;
+        }
+      }
+    }
+
+    bool InteractiveCustomActionCli::parseAction(std::string &line) {
+      if (isBackOption(line)) {
+        return false;
+      }
+
+      auto res = handleParse<std::string>(
+          this, line, actionHandlers_, requestParamsDescriptions_);
+      if (not res.has_value()) {
+        // Continue parsing
+        return true;
```

Comment:
Are you sure it should return `true` when res has no value

Reply:
Yes, user will receive the same message about choice if he/she enters nothing, ```false``` semantically means "go back". Logic the same as for query_cli and transaction_cli.